File: app/src/main/java/net/afterday/compas/core/inventory/Inventory.java
package net.afterday.compas.core.inventory;

import net.afterday.compas.R;
import net.afterday.compas.core.inventory.items.Item;
import net.afterday.compas.core.player.PlayerProps;
import net.afterday.compas.core.serialization.Jsonable;
import net.afterday.compas.persistency.items.ItemDescriptor;

import java.util.List;
import java.util.Map;

/**
 * Created by spaka on 4/18/2018.
 */

public interface Inventory extends Jsonable
{
    int MAX_ARTIFACTS_COUNT = 5;
    Item addItem(ItemDescriptor item, String code);
    boolean hasActiveArmor();
    boolean hasHealthInstant();
    boolean hasRadiationInstant();
    Item consumeArmor(long delta);
    Item getActiveArmor();
    Item getActiveBooster();
    Item getActiveDevice();
    void setPlayerLevel(int level);
    boolean hasActiveBooster();
    boolean hasActiveDevice();
    Item consumeBooster(long delta);
    Item consumeDevice(long delta);
    PlayerProps useItem(Item item, PlayerProps playerProps);
    boolean dropItem(Item item);
    List<Item> getItems();
    List<Item> getBoosters();
    List<Item> getDevices();
    List<Item> getArmors();
    double[] getArtifacts();
}

File: engine/src/main/kotlin/net/afterday/compas/engine/core/inventory/Inventory.kt
package net.afterday.compas.engine.core.inventory

import net.afterday.compas.engine.core.inventory.items.Item
import net.afterday.compas.engine.core.inventory.items.ItemDescriptor
import net.afterday.compas.engine.core.player.PlayerProps
import net.afterday.compas.engine.core.serialization.Jsonable

interface Inventory : Jsonable
{
    fun addItem(item: ItemDescriptor, code: String):Item
    fun useItem(item: Item, playerProps: PlayerProps):PlayerProps
    fun dropItem(item:Item): Boolean

    val items: List<Item>
    val armors: List<Item>
    val boosters: List<Item>
    val devices: List<Item>

    fun setPlayerLevel(level: Int)
    val artifacts: DoubleArray

    fun hasHealthInstant(): Boolean
    fun hasRadiationInstant(): Boolean

}